scripts/build_test.py

```diff
@@ -14,29 +14,313 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import StringIO
+import collections
+import os
+import random
+import re
+import subprocess
+import threading
+
 # pylint: disable=relative-import
 import build
 from core.tests import test_utils
 
 # pylint: enable=relative-import
 
+TEST_DIRECTORY = os.path.join('core', 'tests', 'build', '')
+TEST_SOURCE_DIRECTORY = os.path.join('core', 'tests', 'build_sources')
+
+ASSETS_DEV_DIR = os.path.join(TEST_SOURCE_DIRECTORY, 'assets', '')
+ASSETS_OUT_DIR = os.path.join(TEST_DIRECTORY, 'static', 'assets', '')
+
+EXTENSIONS_DEV_DIR = os.path.join(TEST_SOURCE_DIRECTORY, 'extensions', '')
+
+TEMPLATES_DEV_DIR = os.path.join(TEST_SOURCE_DIRECTORY, 'templates', '')
+
+INVALID_INPUT_FILEPATH = os.path.join(
+    TEST_DIRECTORY, 'invalid', 'path', 'to', 'input.js')
+INVALID_OUTPUT_FILEPATH = os.path.join(
+    TEST_DIRECTORY, 'invalid', 'path', 'to', 'output.js')
+
+EMPTY_DIRECTORY = os.path.join(TEST_DIRECTORY, 'empty', '')
+
+# Override Pylint's protected access rule due to multiple private functions in
+# the file.
+# pylint: disable=protected-access
+
 
 class BuildTests(test_utils.GenericTestBase):
     """Test the build methods."""
 
+    def setUp(self):
+        super(BuildTests, self).setUp()
+        build.safe_delete_directory_tree(TEST_DIRECTORY)
+        build.safe_delete_directory_tree(EMPTY_DIRECTORY)
```

Comment:
Should also do this in tearDown(), right? (Or, could do it in just the relevant tests, too.)